### PR TITLE
Add option to remove all data on uninstall. #771

### PR DESCRIPTION
**Pull request by nosegraze**
#771

### CODE REVIEW

File: uninstall.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	// Clear scheduled cron events.
+	wp_clear_scheduled_hook( 'rcp_expired_users_check' );
+	wp_clear_scheduled_hook( 'rcp_send_expiring_soon_notice' );
+	wp_clear_scheduled_hook( 'rcp_check_member_counts' );
+
+	// Remove all plugin settings.
+	$options = array( 'rcp_settings', 'rcp_db_version', 'rcp_discounts_db_version', 'rcp_payments_db_version', 'rcp_is_installed', 'rcp_version', 'rcp_license_status' );
+	foreach( $options as $option) {
+		delete_option( $option );
```
Comment (pippinsplugins) [inline uninstall.php]: Let's do this as a single query directly to $wpdb for any option `LIKE 'rcp\_%'`

Comment (nosegraze) [inline uninstall.php]: Updated!

File: uninstall.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	$wpdb->query( "DELETE FROM $wpdb->options WHERE option_name LIKE 'rcp\_%'" );
+
+	// Remove all database tables.
+	$wpdb->query( $wpdb->prepare( "DROP TABLE IF EXISTS %s", rcp_get_discounts_db_name() ) );
+	$wpdb->query( $wpdb->prepare( "DROP TABLE IF EXISTS %s", rcp_get_payments_db_name() ) );
+	$wpdb->query( $wpdb->prepare( "DROP TABLE IF EXISTS %s", rcp_get_payment_meta_db_name() ) );
+	$wpdb->query( $wpdb->prepare( "DROP TABLE IF EXISTS %s", rcp_get_level_meta_db_name() ) );
+	$wpdb->query( $wpdb->prepare( "DROP TABLE IF EXISTS %s", rcp_get_levels_db_name() ) );
+
+	// Remove all options.
```
Comment (nosegraze) [inline uninstall.php]: Options are already removed further up.

Comment (mindctrl) [inline uninstall.php]: Ah, didn't see that. But it's not working. I need to run some more tests.

Comment (mindctrl) [inline uninstall.php]: The global $rcp_options isn't available at this time, at least for me, which is why nothing was getting deleted in my tests. Manually getting it with get_option works for me though. Going to tinker with this a little more later tonight.

**Pull request merged**